test/micro/org/openjdk/bench/jdk/incubator/vector/ColumnFilterBenchmark.java

```diff
@@ -0,0 +1,157 @@
+ *
+ *  This code is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ *  version 2 for more details (a copy is included in the LICENSE file that
+ *  accompanied this code).
+ *
+ *  You should have received a copy of the GNU General Public License version
+ *  2 along with this work; if not, write to the Free Software Foundation,
+ *  Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *  Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ *  or visit www.oracle.com if you need additional information or have any
+ *  questions.
+ *
+ */
+
+package org.openjdk.bench.jdk.incubator.vector;
+
+import java.util.concurrent.TimeUnit;
+import java.util.Random;
+import jdk.incubator.vector.*;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+@Fork(jvmArgsPrepend = {"--add-modules=jdk.incubator.vector", "-XX:UseAVX=2"})
+public class ColumnFilterBenchmark {
+    @Param({"1024","2047", "4096"})
+    int size;
+
+    float [] floatinCol;
+    float [] floatoutCol;
+    float fpivot;
+
+    double [] doubleinCol;
+    double [] doubleoutCol;
+    double dpivot;
+
+    int [] intinCol;
+    int [] intoutCol;
+    int ipivot;
+
+    long [] longinCol;
+    long [] longoutCol;
+    long lpivot;
+
+    static final VectorSpecies<Float> fspecies = FloatVector.SPECIES_256;
+    static final VectorSpecies<Double> dspecies = DoubleVector.SPECIES_256;
+    static final VectorSpecies<Integer> ispecies = IntVector.SPECIES_256;
+    static final VectorSpecies<Long> lspecies = LongVector.SPECIES_256;
+
+    @Setup(Level.Trial)
+    public void BmSetup() {
+        Random r = new Random(2048);
+
+        floatinCol = new float[size];
+        floatoutCol = new float[size];
+        fpivot = (float) (size / 2);
+        doubleinCol = new double[size];
+        doubleoutCol = new double[size];
+        dpivot = (double) (size / 2);
+        intinCol = new int[size];
+        intoutCol = new int[size];
+        ipivot = size / 2;
+        longinCol = new long[size];
+        longoutCol = new long[size];
+        lpivot = size / 2;
+
+        for (int i = 4; i < size; i++) {
+            floatinCol[i] = r.nextFloat() * size;
+            doubleinCol[i] = r.nextDouble() * size;
+            intinCol[i] = r.nextInt(size);
+            longinCol[i] = (long)intinCol[i];
+        }
+    }
+
+    @Benchmark
+    public void filterIntColumn() {
+       int i = 0;
+       int j = 0;
+       int endIndex = ispecies.loopBound(size);
+       for (; i < endIndex; i += ispecies.length()) {
+           IntVector vec = IntVector.fromArray(ispecies, intinCol, i);
+           VectorMask<Integer> pred = vec.compare(VectorOperators.GT, ipivot);
+           vec.compress(pred).intoArray(intoutCol, j);
```

Comment:
Could there be equivalent `expand` tests?

Reply:
And what about some result verification? Or is there another test that does that?

Reply:
We do have extensive functional tests for compress/expand APIs in [test/jdk/jdk/incubator/vector](https://github.com/openjdk/jdk/tree/master/test/jdk/jdk/incubator/vector)

Reply:
> Could there be equivalent `expand` tests?

Here are the performance number for existing [VectorAPI JMH micros.](https://github.com/openjdk/panama-vector/tree/vectorIntrinsics/test/micro/org/openjdk/bench/jdk/incubator/vector/operation)

![image](https://github.com/openjdk/jdk/assets/59989778/4b260814-3d3c-4e9b-b81a-61492ea48cce)
![image](https://github.com/openjdk/jdk/assets/59989778/50048281-ad50-44f6-a875-308e02537be2)